CaseTriggerHandler.cls

```diff
@@ -0,0 +1,108 @@
+/******************************************************************
+ * @Name         : CaseTriggerHandler
+ * @Description  : Methods of Contact Trigger
+ * @Created By   : Cagri Kilic
+ * @Created Date : Jan 18, 2023
+ * @Modification Log :
+ ******************************************************************
+ * Version        Developer        Date        Description
+ *------------------------------------------------------------
+ *
+ ******************************************************************/
+public with sharing class CaseTriggerHandler {
+    private static final String RECORD_TYPE_LABEL = 'Person Account';
+    private static final String BRAND_TYPE_1 = 'Option 1';
+    private static final String BRAND_TYPE_2 = 'Option 2';
+    private static final String ORIGIN_TYPE_1 = 'Phone';
+    private static final String ORIGIN_TYPE_2 = 'Email';
+    private static String recordTypeId = Schema.SObjectType.Account.getRecordTypeInfosByName().get(RECORD_TYPE_LABEL).getRecordTypeId();
+    /******************************************************************************
+     * @Name               : addAccountToCase
+     * @Description        : Adds Person Account to Case
+     * @Created By         : Cagri Kilic
+     * @Created Date       : Jan 18, 2023
+     * @Param newCases     : List<Case> - Case list
+     *****************************************************************************/
+    public static void addAccountToCase(List<Case> newCases) {
+        try {
+            List<Account> newAccounts = new List<Account>();
+            Map<String, List<String>> oldCasesMap = new Map<String, List<String>>();
+            List<Case> casesWithoutAccount = new List<Case>();
+            List<Case> oldCases = [
```

Comment:
maybe firstly process new cases and then select existing based on brands/emails ?